streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -122,7 +126,15 @@ public static FileLock lockStateDirectory(File stateDir) throws IOException {
 
     private static FileLock lockStateDirectory(File stateDir, int retry) throws IOException {
         File lockFile = new File(stateDir, ProcessorStateManager.LOCK_FILE_NAME);
-        FileChannel channel = new RandomAccessFile(lockFile, "rw").getChannel();
+        FileChannel channel = null;
+        synchronized (channels) {
+            channel = channels.get(lockFile);
+            if (channel == null) {
+                channel = new RandomAccessFile(lockFile, "rw").getChannel();
+                channels.put(lockFile, channel);
```

Comment:
I think if we use the map to hold all the channels, we should not close channel in this case since the channel might be already in the map from previous invocations.  Leave the channel object wouldn't cause too much problems, there are limited number of locks in the system, usually only in the range of dozen in a normally deployed system (and normal process usually holds 32K file descriptors).